test/prober/configuration.go

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 The Knative Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package prober
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	"knative.dev/eventing/test/lib"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"path"
+	"runtime"
+	"text/template"
+)
+
+const (
+	configName       = "wathola-config"
+	configMountPoint = "/.config/wathola"
+	configFilename   = "config.toml"
+	watholaEventNs   = "com.github.cardil.wathola"
+	healthEndpoint   = "/healthz"
+)
+
+var (
+	eventTypes = []string{"step", "finished"}
+)
+
+func (p *prober) deployConfiguration() {
+	p.annotateNamespace()
+	p.deploySecret()
+	p.deployTriggers()
+}
+
+func (p *prober) annotateNamespace() {
+	ns, err := p.client.Kube.Kube.CoreV1().Namespaces().
+		Get(p.client.Namespace, metav1.GetOptions{})
+	lib.NoError(err)
+	ns.Labels = map[string]string{
+		"knative-eventing-injection": "enabled",
+	}
+	_, err = p.client.Kube.Kube.CoreV1().Namespaces().
+		Update(ns)
+	lib.NoError(err)
+}
+
+func (p *prober) deploySecret() {
+	name := configName
+	p.log.Infof("Deploying secret: %v", name)
+
+	configData := p.compileTemplate(configFilename)
+	data := make(map[string]string, 0)
+	data[configFilename] = configData
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		StringData: data,
+		Type:       "Opaque",
+	}
+	_, err := p.client.Kube.Kube.CoreV1().Secrets(p.config.Namespace).
+		Create(secret)
+	lib.NoError(err)
+}
+
+func (p *prober) deployTriggers() {
+	for _, eventType := range eventTypes {
```

Comment:
Can we create these triggers like what we are doing in e2e tests - https://github.com/knative/eventing/blob/62fa78e8a14626a37f6e644cfbd97e657261422f/test/e2e/helpers/broker_event_transformation_test_helper.go#L77-L82 ?

Reply:
They target knative service, so no. Unless we decide to vendor `knative/serving`.

Reply:
Can you use https://github.com/knative/eventing/blob/8a92a8c91447a6ff71e4be97a42c548790d52a22/test/lib/resources/eventing.go#L180?

(This function has a bug that needs a fix though... But just the idea.)

Reply:
I used that.